### DISCUSSION TITLE
i want to know to where this endpoint is called

### DISCUSSION THREAD

**Discussion by AstinOfficial** [appium/appium#20047] [General]
async function getDisplayDensity() {
    return /** @type {number} */ (await this.uiautomator2.jwproxy.command('/appium/device/display_density', 'GET', {}));
}

this is the code in uiautomator 2 driver of appium  

i want to know to where this endpoint is called

**Comment by KazuCocoa** [comment]
It goes to https://github.com/appium/appium-uiautomator2-driver/blob/ccc7d54fc2eca6d24fbdc646fae9f07361bb2e8e/lib/commands/viewport.js#L77

**Comment by mykola-mokhnach** [comment]
It calls the corresponding server API: https://github.com/appium/appium-uiautomator2-server/blob/a7190b06ffcf19f390a5f49b82880a4445fdcfdb/app/src/main/java/io/appium/uiautomator2/server/AppiumServlet.java#L181
